projects/court-case-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/proxy/CommunityApiController.kt

```diff
@@ -198,7 +220,8 @@ class CommunityApiController(
         val reports = personList.content.flatMap { person ->
             val convictionId = person.events.filter { it.disposal != null }.maxOfOrNull { it.id }
             val nsiCodes = person.nsis.filter { it.eventId == convictionId }.map { it.type.code.trim() }
-            runAll(person.crn, convictionId, nsiCodes, compare, headers).stream()
+            val nsiId = person.nsis.firstOrNull()?.id
+            runAll(person.crn, convictionId, nsiCodes, nsiId, compare, headers).stream()
                 .map(CompletableFuture<CompareReport>::join)
                 .collect(Collectors.toList())
```

Comment:
Not really a change in this PR, but just spotted you don't need to convert to a stream here.  This would also work:
```suggestion
            runAll(person.crn, convictionId, nsiCodes, nsiId, compare, headers).map { it.join() }
```

Also I think this is running in sequence and not in parallel?  Think you need to use `CompletableFuture.allOf` to get them to run in parallel:
```suggestion
            CompletableFuture.allOf(person.crn, convictionId, nsiCodes, nsiId, compare, headers).join()
            return listOf(person.crn, convictionId, nsiCodes, nsiId, compare, headers).map { it.get() }
```

Reply:
Changed